lib/codemirror.js

```diff
@@ -587,8 +587,11 @@
   function alignHorizontally(cm) {
     var display = cm.display, view = display.view;
     if (!display.alignWidgets && (!display.gutters.firstChild || !cm.options.fixedGutter)) return;
-    var comp = compensateForHScroll(display) - display.scroller.scrollLeft + cm.doc.scrollLeft;
+
+    var editableOffset = cm.isReadOnly() ? display.scroller.scrollLeft : cm.doc.scrollLeft;
```

Comment:
The DOM for a read-only editor is (supposed to be) structurally the same as that for an editable one. What is the reasoning behind this conditional?